Neos.ContentRepository.Core/Classes/CommandHandler/CommandInterface.php

```diff
@@ -5,10 +5,14 @@
 namespace Neos\ContentRepository\Core\CommandHandler;
 
 /**
- * Common (marker) interface for all commands of the Content Repository
+ * Common interface for all commands of the Content Repository
  *
  * @internal because extra commands are no extension point
  */
 interface CommandInterface
 {
+    /**
+     * @param array<string,mixed> $array
+     */
+    public static function fromArray(array $array): self;
```

Comment:
> this whole commands that are serialised, vs commands that are not a little more confusing

Confusing because they both have a `fromArray()` method instead of just _some_ commands before?
Anyways, with https://github.com/neos/neos-development-collection/pull/5348 that won't be an issue – or did I get you wrong?

> this fromArray is not needed for the core but just for tests

I don't think so. Being able to construct any command from an array will be very useful in userland, i.e.:
```php
public function someAction(array $command) {
    $this->cr->handle(SomeCommand::fromArray($command));
    // ...
}
```

Reply:
Mmm, yeah that is.... meh

Reply:
Ok fair if we see this as a real userland feature, I am on board, as long as we do not internal it as just for tests 😆

Reply:
most commands already had it anyways

Reply:
Oh I know, is why I said in the discussion about serialized commands and interfaces that I would prefer this interface to have _something_ implemented, but I thought the fromArray was purely internal.